java/client/src/test/java/glide/api/RedisClientTest.java

```diff
@@ -6207,4 +6211,102 @@ public void lcs_with_len_option() {
         assertEquals(testResponse, response);
         assertEquals(value, payload);
     }
+
+    @SneakyThrows
+    @Test
+    public void dump_returns_success() {
+        // setup
+        byte[] key = "testKey".getBytes();
+        byte[] value = "value".getBytes();
+        List<byte[]> arguments = List.of(key);
+
+        CompletableFuture<byte[]> testResponse = new CompletableFuture<>();
+        testResponse.complete(value);
+
+        // match on protobuf request
+        when(commandManager.<byte[]>submitNewCommand(eq(Dump), eq(arguments), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<byte[]> response = service.dump(key);
+        byte[] payload = response.get();
+
+        // verify
+        assertEquals(testResponse, response);
+        assertEquals(value, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void restore_noOptions_returns_success() {
```

Comment:
nit: since the next one is named restore_withoptions this one can just be restore?